ui/dashboardApp/index.ts

```diff
@@ -34,7 +60,10 @@ async function main() {
 
   singleSpa.registerApplication(
     'layout',
-    AppRegistry.newReactSpaApp(() => LayoutMain, 'root'),
+    AppRegistry.newReactSpaApp(
+      () => (appOptions.hideNav ? LayoutFull : LayoutMain),
+      'root'
```

Comment:
We can pass the config to each App (modify AppRegistry). Then we even don't need to introduce a new Layout.